cdap-data-fabric/src/main/java/co/cask/cdap/data/runtime/DataSetsModules.java

```diff
@@ -41,6 +44,10 @@
 
   @Override
   public Module getInMemoryModules() {
+    return getInMemoryModules(false);
+  }
+
+  public Module getInMemoryModules(final boolean kafkaRequired) {
```

Comment:
seems like the boolean should be named publishRequired. Whether Kafka is required is a consequence of that.